lib/badge-data.js

```diff
@@ -107,8 +107,14 @@ function makeBadgeData(defaultLabel, overrides) {
     logoWidth: +overrides.logoWidth,
     links: toArray(overrides.link),
     // Scoutcamp sometimes turns these into numbers.
-    colorA: `${overrides.colorA}`,
-    colorB: `${overrides.colorB}`,
+    colorA:
+      typeof overrides.colorA === 'number'
+        ? `${overrides.colorA}`
+        : overrides.colorA,
+    colorB:
+      typeof overrides.colorB === 'number'
+        ? `${overrides.colorB}`
+        : overrides.colorB,
```

Comment:
This was the real issue. It was coercing `undefined` to `'undefined'` which was then taking precedence over any value of `colorscheme`.